pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -89,35 +89,58 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundHTTPRouteConfig(env *mo
 // buildSidecarOutboundHTTPRouteConfig builds an outbound HTTP Route for sidecar.
 // Based on port, will determine all virtual hosts that listen on the port.
 func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPRouteConfig(env *model.Environment, node *model.Proxy, push *model.PushContext,
-	proxyInstances []*model.ServiceInstance, services []*model.Service, routeName string) *xdsapi.RouteConfiguration {
+	proxyInstances []*model.ServiceInstance, routeName string) *xdsapi.RouteConfiguration {
 
 	listenerPort := 0
-	if routeName != RDSHttpProxy {
-		var err error
-		listenerPort, err = strconv.Atoi(routeName)
-		if err != nil {
+	var err error
+	listenerPort, err = strconv.Atoi(routeName)
+	if err != nil {
+		// we have a port whose name is http_proxy or unix:///foo/bar
```

Comment:
What is this ? Why would a route be named unix:///foo/bar ???

Reply:
Please move the UDS implementation to separate PR, if you keep adding features it'll be harder to review and merge this.